event_reporter/reporter/event_payload.go

```diff
@@ -20,205 +20,237 @@ import (
-			errors = append(errors, parseAggregativeHealthErrors(rs, apptree, false)...)
-		}
+	if rr.rs.Health != nil {
+		source.HealthStatus = (*string)(&rr.rs.Health.Status)
+		source.HealthMessage = &rr.rs.Health.Message
 	}
 
 	payload := events.EventPayload{
 		Timestamp:   appEventProcessingStartedAt,
 		Object:      object,
 		Source:      &source,
-		Errors:      errors,
+		Errors:      getResourceEventPayloadErrors(rr, reportedEntityParentApp),
 		AppVersions: applicationVersionsEvents,
 	}
 
-	logCtx.Infof("AppVersion before encoding: %v", utils.SafeString(payload.AppVersions.AppVersion))
+	if payload.AppVersions != nil {
+		logCtx.Infof("AppVersion before encoding: %v", utils.SafeString(payload.AppVersions.AppVersion))
+	}
 
 	payloadBytes, err := json.Marshal(&payload)
 	if err != nil {
-		return nil, fmt.Errorf("failed to marshal payload for resource %s/%s: %w", rs.Namespace, rs.Name, err)
+		return nil, fmt.Errorf("failed to marshal payload for resource %s/%s: %w", rr.rs.Namespace, rr.rs.Name, err)
 	}
 
 	return &events.Event{Payload: payloadBytes}, nil
 }
 
+func getResourceEventPayloadErrors(
+	rr *ReportedResource,
+	reportedEntityParentApp *ReportedEntityParentApp,
+) []*events.ObjectError {
+	var errors []*events.ObjectError
+
+	if reportedEntityParentApp.app.Status.OperationState != nil {
+		errors = append(errors, parseResourceSyncResultErrors(rr.rs, reportedEntityParentApp.app.Status.OperationState)...)
+	}
+
+	// parent application not include errors in application originally was created with broken state, for example in destination missed namespace
+	if rr.rsAsAppInfo != nil && rr.rsAsAppInfo.app != nil {
+		if rr.rsAsAppInfo.app.Status.OperationState != nil {
+			errors = append(errors, parseApplicationSyncResultErrors(rr.rsAsAppInfo.app.Status.OperationState)...)
+		}
+
+		if rr.rsAsAppInfo.app.Status.Conditions != nil {
+			errors = append(errors, parseApplicationSyncResultErrorsFromConditions(rr.rsAsAppInfo.app.Status)...)
+		}
+
+		errors = append(errors, parseAggregativeHealthErrorsOfApplication(rr.rsAsAppInfo.app, reportedEntityParentApp.appTree)...)
+	}
+
+	if rr.rs.Health != nil && rr.rs.Health.Status != health.HealthStatusHealthy {
+		errors = append(errors, parseAggregativeHealthErrors(rr.rs, reportedEntityParentApp.appTree, false)...)
+	}
+
+	return errors
+}
+
+func useCompiledManifestAsEventObject(
+	rr *ReportedResource,
+) ([]byte, error) {
+	// no actual state, use desired state as event object
+	unstructuredWithNamespace, err := utils.AddDestNamespaceToManifest([]byte(rr.desiredState.CompiledManifest), rr.rs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add destination namespace to manifest: %w", err)
+	}
+
+	return addCommitDetailsToUnstructured(unstructuredWithNamespace, rr)
+}
+
+func buildEventObjectAsLiveAndCompiledManifestsEmpty(
+	rr *ReportedResource,
+) ([]byte, error) {
+	// no actual or desired state, don't send event
+	u := &unstructured.Unstructured{}
+	apiVersion := rr.rs.Version
```

Comment:
create function inside ReportedResource that will build you api version, like rr.rs.GetApiVersion